src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -302,6 +373,20 @@ public void Dispose()
                     _fakeMemoryOwner = null;
                 }
 
+                // Call dispose on any memory that wasn't written.
+                if (_completedSegments != null)
+                {
+                    foreach (var segment in _completedSegments)
+                    {
+                        segment.Return();
+                    }
+                }
+
+                if (_currentSegmentOwner != null)
+                {
+                    _currentSegmentOwner.Dispose();
```

Comment:
Why set _currentSegmentOwner to null in WriteDataWrittenBeforeHeaders but not here?